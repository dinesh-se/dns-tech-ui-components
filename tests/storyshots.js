import path from 'path'
import initStoryshots from '@storybook/addon-storyshots'
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer'

const getMatchOptions = () => ({
  failureThreshold: 0.2,
  failureThresholdType: 'percent',
})

const beforeScreenshot = (page) => {
  return new Promise((resolve) =>
    setTimeout(() => {
      page.setViewport({
        width: 800,
        height: 600,
        deviceScaleFactor: 1,
      })
      resolve()
    })
  )
}

initStoryshots({
  suite: 'Image storyshots',
  test: imageSnapshot({
    storybookUrl: `file:///${path.join(__dirname, '..', 'storybook-static')}`,
    beforeScreenshot,
    getMatchOptions,
  }),
})
