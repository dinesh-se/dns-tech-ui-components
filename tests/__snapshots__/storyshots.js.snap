// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Button Button Groups 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <div
    className="btn-group btn-group-block"
  >
    <button
      className="btn"
      disabled={false}
      type="button"
    >
      first button
    </button>
    <button
      className="btn"
      disabled={false}
      type="button"
    >
      second button
    </button>
    <button
      className="btn"
      disabled={false}
      type="button"
    >
      third button
    </button>
  </div>
</div>
`;

exports[`Storyshots Components/Button Disabled 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn"
    disabled={true}
    tabIndex={-1}
    type="button"
  >
    default
  </button>
  <button
    className="btn btn-primary"
    disabled={true}
    tabIndex={-1}
    type="button"
  >
    primary
  </button>
  <button
    className="btn btn-link"
    disabled={true}
    tabIndex={-1}
    type="button"
  >
    link
  </button>
</div>
`;

exports[`Storyshots Components/Button Loading 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn loading"
    disabled={false}
    type="button"
  >
    default
  </button>
  <button
    className="btn btn-primary loading"
    disabled={false}
    type="button"
  >
    primary
  </button>
  <button
    className="btn btn-link loading"
    disabled={false}
    type="button"
  >
    link
  </button>
</div>
`;

exports[`Storyshots Components/Button Playground 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn"
    disabled={false}
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Storyshots Components/Button Sizes 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn btn-sm"
    disabled={false}
    type="button"
  >
    small
  </button>
  <button
    className="btn"
    disabled={false}
    type="button"
  >
    medium
  </button>
  <button
    className="btn btn-lg"
    disabled={false}
    type="button"
  >
    large
  </button>
  <button
    className="btn btn-block"
    disabled={false}
    type="button"
  >
    block
  </button>
</div>
`;

exports[`Storyshots Components/Button States 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn btn-success"
    disabled={false}
    type="button"
  >
    success
  </button>
  <button
    className="btn btn-error"
    disabled={false}
    type="button"
  >
    error
  </button>
</div>
`;

exports[`Storyshots Components/Button Variants 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <button
    className="btn"
    disabled={false}
    type="button"
  >
    default
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    type="button"
  >
    primary
  </button>
  <button
    className="btn btn-link"
    disabled={false}
    type="button"
  >
    link
  </button>
</div>
`;

exports[`Storyshots Components/Table Default 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <div
    className="table"
  >
    <table
      className="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th>
            Header 1
          </th>
          <th>
            Header 2
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Column 11
          </td>
          <td>
            Column 12
          </td>
        </tr>
        <tr>
          <td>
            Column 21
          </td>
          <td>
            Column 22
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Storyshots Components/Table No Data 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <div
    className="table"
  >
    <table
      className="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th>
            Header 1
          </th>
          <th>
            Header 2
          </th>
        </tr>
      </thead>
    </table>
    <div
      className="sc-gsTCUz cMktrB"
    >
      No Data
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Table Play Ground 1`] = `
<div
  className="sc-bdfBwQ dICnXw"
>
  <div
    className="table"
  >
    <table
      className="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th>
            Header 1
          </th>
          <th>
            Header 2
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Column 11
          </td>
          <td>
            Column 12
          </td>
        </tr>
        <tr>
          <td>
            Column 21
          </td>
          <td>
            Column 22
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
